Lint检查失败: 命令系统代码存在 39 个 ESLint 错误

## 测试输出错误信息

运行 \`npm run lint\` 时发现 **39 个错误** 和 65 个警告。

### 主要错误分布

| 文件 | 错误数 | 错误类型 |
|------|--------|----------|
| `src/nodes/commands/builtin-commands.ts` | 12 | `require-await` - async 方法没有 await |
| `src/nodes/commands/command-registry.test.ts` | 22 | `require-await` - async 方法没有 await |
| `src/nodes/commands/command-registry.ts` | 3 | `no-duplicate-imports`, `curly` |
| `src/channels/feishu-channel.ts` | 2 | `prefer-destructuring` |
| `src/platforms/feishu/welcome-service.ts` | 1 | `require-await` |

### 具体错误示例

**1. require-await 错误 (最常见)**
```
src/nodes/commands/builtin-commands.ts
  19:3  error  Async method 'execute' has no 'await' expression  require-await
  36:3  error  Async method 'execute' has no 'await' expression  require-await
  ...
```

**2. no-duplicate-imports 错误**
```
src/nodes/commands/command-registry.ts
  13:1  error  './types.js' import is duplicated  no-duplicate-imports
```

**3. curly 错误**
```
src/nodes/commands/command-registry.ts
  125:18  error  Expected { after 'if' condition    curly
  128:34  error  Expected { after 'if' condition    curly
```

**4. prefer-destructuring 错误**
```
src/channels/feishu-channel.ts
  637:11  error  Use object destructuring  prefer-destructuring
  659:11  error  Use object destructuring  prefer-destructuring
```

## 根本原因分析

这些错误来自于 PR #531 (fix/issue-463-help-system-v2) 新增的命令系统代码：

1. **require-await 错误**: 命令的 `execute` 方法被声明为 `async`，但实现中没有使用 `await`。应该移除 `async` 关键字。

2. **no-duplicate-imports 错误**: `command-registry.ts` 中从 `./types.js` 重复导入。

3. **curly 错误**: if 语句没有使用花括号。

4. **prefer-destructuring 错误**: `feishu-channel.ts` 中应使用对象解构。

## 相关提交

- 42dea7a Merge pull request #531 from hs3180/fix/issue-463-help-system-v2
- 1acb0f8 feat(help): add DI-based command system with welcome messages (Issue #463)
- d88982a feat(help): add DI-based command system with welcome messages (Issue #463)

## 建议修复

1. 移除不需要 `async` 的方法的 `async` 关键字
2. 合并重复的 import 语句
3. 为 if 语句添加花括号
4. 使用对象解构替代属性访问

🤖 由 CI Monitor 自动检测

--- a/src/channels/feishu-channel.ts
+++ b/src/channels/feishu-channel.ts
@@ -634,7 +634,7 @@ export class FeishuChannel extends BaseChannel<FeishuChannelConfig> {
       return;
     }
 
-    const event = data.event;
+    const { event } = data;
     if (!event?.user?.open_id) {
       logger.debug('P2P chat entered event missing user info');
       return;
@@ -656,7 +656,7 @@ export class FeishuChannel extends BaseChannel<FeishuChannelConfig> {
       return;
     }
 
-    const event = data.event;
+    const { event } = data;
     if (!event?.chat_id || !event?.members || event.members.length === 0) {
       logger.debug('Chat member added event missing required fields');
       return;

--- a/src/nodes/commands/builtin-commands.ts
+++ b/src/nodes/commands/builtin-commands.ts
@@ -16,7 +16,7 @@ export class ResetCommand implements Command {
   readonly category = 'session' as const;
   readonly description = '重置对话';
 
-  async execute(_context: CommandContext): Promise<CommandResult> {
+  execute(_context: CommandContext): CommandResult {
     // Actual reset is handled by PrimaryNode
     return {
       success: true,
@@ -33,7 +33,7 @@ export class StatusCommand implements Command {
   readonly category = 'session' as const;
   readonly description = '查看状态';
 
-  async execute(_context: CommandContext): Promise<CommandResult> {
+  execute(_context: CommandContext): CommandResult {
     // Actual status is handled by PrimaryNode
     return {
       success: true,
@@ -56,7 +56,7 @@ export class HelpCommand implements Command {
     this.generateHelpText = generateHelpText;
   }
 
-  async execute(_context: CommandContext): Promise<CommandResult> {
+  execute(_context: CommandContext): CommandResult {
     return {
       success: true,
       message: this.generateHelpText(),
@@ -72,7 +72,7 @@ export class ListNodesCommand implements Command {
   readonly category = 'node' as const;
   readonly description = '列出执行节点';
 
-  async execute(_context: CommandContext): Promise<CommandResult> {
+  execute(_context: CommandContext): CommandResult {
     // Actual implementation is handled by PrimaryNode
     return {
       success: true,
@@ -90,7 +90,7 @@ export class SwitchNodeCommand implements Command {
   readonly description = '切换执行节点';
   readonly usage = 'switch-node <nodeId>';
 
-  async execute(context: CommandContext): Promise<CommandResult> {
+  execute(context: CommandContext): CommandResult {
     if (context.args.length === 0) {
       return {
         success: false,
@@ -113,7 +113,7 @@ export class RestartCommand implements Command {
   readonly category = 'node' as const;
   readonly description = '重启服务';
 
-  async execute(_context: CommandContext): Promise<CommandResult> {
+  execute(_context: CommandContext): CommandResult {
     // Actual implementation is handled by PrimaryNode
     return {
       success: true,
@@ -131,7 +131,7 @@ export class CreateGroupCommand implements Command {
   readonly description = '创建群';
   readonly usage = 'create-group <name> <members>';
 
-  async execute(context: CommandContext): Promise<CommandResult> {
+  execute(context: CommandContext): CommandResult {
     if (context.args.length < 2) {
       return {
         success: false,
@@ -155,7 +155,7 @@ export class AddMemberCommand implements Command {
   readonly description = '添加成员';
   readonly usage = 'add-member <groupId> <member>';
 
-  async execute(context: CommandContext): Promise<CommandResult> {
+  execute(context: CommandContext): CommandResult {
     if (context.args.length < 2) {
       return {
         success: false,
@@ -179,7 +179,7 @@ export class RemoveMemberCommand implements Command {
   readonly description = '移除成员';
   readonly usage = 'remove-member <groupId> <member>';
 
-  async execute(context: CommandContext): Promise<CommandResult> {
+  execute(context: CommandContext): CommandResult {
     if (context.args.length < 2) {
       return {
         success: false,
@@ -203,7 +203,7 @@ export class ListMemberCommand implements Command {
   readonly description = '列出成员';
   readonly usage = 'list-member <groupId>';
 
-  async execute(context: CommandContext): Promise<CommandResult> {
+  execute(context: CommandContext): CommandResult {
     if (context.args.length < 1) {
       return {
         success: false,
@@ -226,7 +226,7 @@ export class ListGroupCommand implements Command {
   readonly category = 'group' as const;
   readonly description = '列出群';
 
-  async execute(_context: CommandContext): Promise<CommandResult> {
+  execute(_context: CommandContext): CommandResult {
     // Actual implementation is handled by PrimaryNode
     return {
       success: true,
@@ -244,7 +244,7 @@ export class DissolveGroupCommand implements Command {
   readonly description = '解散群';
   readonly usage = 'dissolve-group <groupId>';
 
-  async execute(context: CommandContext): Promise<CommandResult> {
+  execute(context: CommandContext): CommandResult {
     if (context.args.length < 1) {
       return {
         success: false,

--- a/src/nodes/commands/command-registry.test.ts
+++ b/src/nodes/commands/command-registry.test.ts
@@ -27,7 +27,7 @@ describe('CommandRegistry', () => {
         name: 'test',
         description: 'Test command',
         category: 'session',
-        execute: async () => ({ success: true, message: 'Test' }),
+        execute: () => ({ success: true, message: 'Test' }),
       };
 
       registry.register(cmd);
@@ -42,13 +42,13 @@ describe('CommandRegistry', () => {
         name: 'test',
         description: 'First',
         category: 'session',
-        execute: async () => ({ success: true }),
+        execute: () => ({ success: true }),
       };
       const cmd2: Command = {
         name: 'test',
         description: 'Second',
         category: 'session',
-        execute: async () => ({ success: true }),
+        execute: () => ({ success: true }),
       };
 
       registry.register(cmd1);
@@ -61,8 +61,8 @@ describe('CommandRegistry', () => {
   describe('registerAll', () => {
     it('should register multiple commands', () => {
       const commands: Command[] = [
-        { name: 'cmd1', description: 'Command 1', category: 'session', execute: async () => ({ success: true }) },
-        { name: 'cmd2', description: 'Command 2', category: 'group', execute: async () => ({ success: true }) },
+        { name: 'cmd1', description: 'Command 1', category: 'session', execute: () => ({ success: true }) },
+        { name: 'cmd2', description: 'Command 2', category: 'group', execute: () => ({ success: true }) },
       ];
 
       registry.registerAll(commands);
@@ -75,8 +75,8 @@ describe('CommandRegistry', () => {
   describe('getAll', () => {
     it('should return all enabled commands', () => {
       const commands: Command[] = [
-        { name: 'enabled', description: 'Enabled', category: 'session', execute: async () => ({ success: true }) },
-        { name: 'disabled', description: 'Disabled', category: 'session', enabled: false, execute: async () => ({ success: true }) },
+        { name: 'enabled', description: 'Enabled', category: 'session', execute: () => ({ success: true }) },
+        { name: 'disabled', description: 'Disabled', category: 'session', enabled: false, execute: () => ({ success: true }) },
       ];
 
       registry.registerAll(commands);
@@ -90,8 +90,8 @@ describe('CommandRegistry', () => {
   describe('getByCategory', () => {
     it('should filter commands by category', () => {
       const commands: Command[] = [
-        { name: 'session-cmd', description: 'Session', category: 'session', execute: async () => ({ success: true }) },
-        { name: 'group-cmd', description: 'Group', category: 'group', execute: async () => ({ success: true }) },
+        { name: 'session-cmd', description: 'Session', category: 'session', execute: () => ({ success: true }) },
+        { name: 'group-cmd', description: 'Group', category: 'group', execute: () => ({ success: true }) },
       ];
 
       registry.registerAll(commands);
@@ -105,9 +105,9 @@ describe('CommandRegistry', () => {
   describe('getActiveCategories', () => {
     it('should return categories in correct order', () => {
       const commands: Command[] = [
-        { name: 'schedule-cmd', description: 'Schedule', category: 'schedule', execute: async () => ({ success: true }) },
-        { name: 'session-cmd', description: 'Session', category: 'session', execute: async () => ({ success: true }) },
-        { name: 'group-cmd', description: 'Group', category: 'group', execute: async () => ({ success: true }) },
+        { name: 'schedule-cmd', description: 'Schedule', category: 'schedule', execute: () => ({ success: true }) },
+        { name: 'session-cmd', description: 'Session', category: 'session', execute: () => ({ success: true }) },
+        { name: 'group-cmd', description: 'Group', category: 'group', execute: () => ({ success: true }) },
       ];
 
       registry.registerAll(commands);
@@ -120,9 +120,9 @@ describe('CommandRegistry', () => {
   describe('generateHelpText', () => {
     it('should generate formatted help text', () => {
       const commands: Command[] = [
-        { name: 'reset', description: '重置对话', category: 'session', execute: async () => ({ success: true }) },
-        { name: 'status', description: '查看状态', category: 'session', execute: async () => ({ success: true }) },
-        { name: 'create-group', description: '创建群', usage: 'create-group <name> <members>', category: 'group', execute: async () => ({ success: true }) },
+        { name: 'reset', description: '重置对话', category: 'session', execute: () => ({ success: true }) },
+        { name: 'status', description: '查看状态', category: 'session', execute: () => ({ success: true }) },
+        { name: 'create-group', description: '创建群', usage: 'create-group <name> <members>', category: 'group', execute: () => ({ success: true }) },
       ];
 
       registry.registerAll(commands);
@@ -138,8 +138,8 @@ describe('CommandRegistry', () => {
 
     it('should not include disabled commands', () => {
       const commands: Command[] = [
-        { name: 'enabled', description: 'Enabled', category: 'session', execute: async () => ({ success: true }) },
-        { name: 'disabled', description: 'Disabled', category: 'session', enabled: false, execute: async () => ({ success: true }) },
+        { name: 'enabled', description: 'Enabled', category: 'session', execute: () => ({ success: true }) },
+        { name: 'disabled', description: 'Disabled', category: 'session', enabled: false, execute: () => ({ success: true }) },
       ];
 
       registry.registerAll(commands);
@@ -153,7 +153,7 @@ describe('CommandRegistry', () => {
   describe('generateWelcomeMessage', () => {
     it('should generate welcome message with help', () => {
       const commands: Command[] = [
-        { name: 'reset', description: '重置对话', category: 'session', execute: async () => ({ success: true }) },
+        { name: 'reset', description: '重置对话', category: 'session', execute: () => ({ success: true }) },
       ];
 
       registry.registerAll(commands);
@@ -171,7 +171,7 @@ describe('CommandRegistry', () => {
         name: 'test',
         description: 'Test',
         category: 'session',
-        execute: async (ctx: CommandContext) => ({
+        execute: (ctx: CommandContext) => ({
           success: true,
           message: `Executed with args: ${ctx.args.join(', ')}`,
         }),
@@ -206,7 +206,7 @@ describe('CommandRegistry', () => {
         description: 'Disabled',
         category: 'session',
         enabled: false,
-        execute: async () => ({ success: true }),
+        execute: () => ({ success: true }),
       };
 
       registry.register(cmd);
@@ -225,7 +225,7 @@ describe('CommandRegistry', () => {
         name: 'error-cmd',
         description: 'Error',
         category: 'session',
-        execute: async () => {
+        execute: () => {
           throw new Error('Test error');
         },
       };

--- a/src/nodes/commands/command-registry.ts
+++ b/src/nodes/commands/command-registry.ts
@@ -9,8 +9,7 @@
  * Issue #463: 帮助消息系统 - 入群/私聊引导 + 指令注册
  */
 
-import type { Command, CommandCategory, CommandContext, CommandResult } from './types.js';
-import { CATEGORY_CONFIG } from './types.js';
+import { CATEGORY_CONFIG, type Command, type CommandCategory, type CommandContext, type CommandResult } from './types.js';
 
 /**
  * Command Registry - Manages control command instances.
@@ -122,10 +121,14 @@ export class CommandRegistry {
 
     for (const category of categories) {
       const info = CATEGORY_CONFIG[category];
-      if (!info) continue;
+      if (!info) {
+        continue;
+      }
 
       const commands = this.getByCategory(category);
-      if (commands.length === 0) continue;
+      if (commands.length === 0) {
+        continue;
+      }
 
       lines.push(`${info.emoji} ${info.label}：`);
 

--- a/src/nodes/commands/types.ts
+++ b/src/nodes/commands/types.ts
@@ -71,9 +71,9 @@ export interface Command {
    * Execute the command.
    *
    * @param context - Command execution context
-   * @returns Command execution result
+   * @returns Command execution result (can be sync or async)
    */
-  execute(context: CommandContext): Promise<CommandResult>;
+  execute(context: CommandContext): CommandResult | Promise<CommandResult>;
 }
 
 /**

--- a/src/platforms/feishu/welcome-service.ts
+++ b/src/platforms/feishu/welcome-service.ts
@@ -110,7 +110,7 @@ export class WelcomeService {
    * Handle P2P chat entered event.
    * This is called when a user starts a private chat with the bot.
    */
-  async handleP2PChatEntered(chatId: string): Promise<boolean> {
+  handleP2PChatEntered(chatId: string): Promise<boolean> {
     return this.handleFirstPrivateChat(chatId);
   }
 

